lib/elastomer/client/docs.rb

```diff
@@ -265,12 +272,44 @@ def count( query, params = nil )
       #
       # See http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/docs-delete-by-query.html
       #
-      # Returns the response body as a hash
+      # Returns a Hash of statistics about the delete operations
       def delete_by_query( query, params = nil )
         query, params = extract_params(query) if params.nil?
 
-        response = client.delete '/{index}{/type}/_query', update_params(params, :body => query, :action => 'docs.delete_by_query')
-        response.body
+        responses = []
+        accumulate = lambda { |response| responses << response unless response == nil }
+
+        accumulate.call(bulk do |b|
+          scan(query, params).each_document do |hit|
+            accumulate.call(b.delete(_id: hit["_id"]))
+          end
+        end)
+
+        response_items = responses.flat_map { |r| r["items"].map { |i| i["delete"] } }
+
+        is_ok = lambda { |status| status.between?(200, 299) }
+
+        categorize = lambda do |items|
+          {
+            "found" => items.count { |i| i["found"] },
+            "deleted" => items.count { |i| is_ok.call(i["status"]) },
+            "missing" => items.count { |i| !i["found"] },
+            "failed" => items.count { |i| i["found"] && !is_ok.call(i["status"]) },
+          }
+        end
+
+        indices = Hash[response_items
+          .map { |i| { i["_index"] => [i.tap { |ip| ip.delete("_index") }] } }
+          .reduce({}) { |acc, i| acc.merge(i) { |_, old, new| old + new } }
```

Comment:
[Enumerable#group_by](http://ruby-doc.org/core-1.9.3/Enumerable.html#method-i-group_by) might be useful. Could possibly replace this line and the previous.